inlang/source-code/sdk/src/resolve-modules/cache.ts

```diff
@@ -2,10 +2,15 @@ import type { NodeishFilesystemSubset } from "@inlang/plugin"
 import { type Result, tryCatch } from "@inlang/result"
 
 function escape(url: string) {
-	// collect the bytes of the UTF-8 representation & hex encode
+	// collect the bytes of the UTF-8 representation
 	const bytes = new TextEncoder().encode(url)
-	const escaped = [...bytes].map((byte) => byte.toString(16).padStart(2, "0")).join("")
-	return escaped
+
+	// 32-bit FNV1a hash to make the file-names shorter
+	// https://en.wikipedia.org/wiki/FNV-1a
+	const hash = bytes.reduce((hash, byte) => (hash * 0x01000193) ^ byte, 0x811c9dc5)
```

Comment:
I'm not a mathematician - If you validate this say against https://github.com/sindresorhus/fnv1a to make sure it captures the same algorithm, then we can merge it.

Reply:
I upgraded to a 64 bit version for more hash space. 

The implementations actually weren't equivalent. Mine didn't discard the bits past 32 bits, so there was some extra junk in the binary representation. The upgraded 64bit one is equivalent.